wormhole/file_transport.go

```diff
@@ -410,23 +435,20 @@ func (t *fileTransport) makeTransitMsg() (*transitMsg, error) {
 		}
 	}
 
-	if t.relayConn != nil {
-		var relayType string
-		switch t.relayURL.Scheme {
-		case "tcp":
-			relayType = "direct-tcp-v1"
-		case "ws":
-			relayType = "websocket-v1"
-		case "wss":
-			relayType = "websocket-v1"
-		default:
-			return nil, fmt.Errorf("%w: '%s'", UnsupportedProtocolErr, t.relayURL.Scheme)
-		}
-		if relayType == "direct-tcp-v1" {
-			var port, err = strconv.Atoi(t.relayURL.Port())
-			if err != nil {
-				return nil, fmt.Errorf("invalid port")
-			}
+	var relayType string
+	switch t.relayURL.Scheme {
+	case "tcp":
+		relayType = "direct-tcp-v1"
+	case "ws":
+		relayType = "websocket-v1"
+	case "wss":
+		relayType = "websocket-v1"
+	default:
+		return nil, fmt.Errorf("%w: '%s'", UnsupportedProtocolErr, t.relayURL.Scheme)
+	}
+	if relayType == "direct-tcp-v1" {
+		var port, err = strconv.Atoi(t.relayURL.Port())
+		if err == nil {
```

Comment:
May be we should handle the `err != nil` case also here and return some error value like in the original code?